doc/mpls/MPLS_hld.md

```diff
@@ -0,0 +1,390 @@
+To facilitate weighted ECMP route functionality, the following update was made to the SAI Next Hop Group API:
+```
+    /**
+     * @brief Member weights
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 1
+     */
+    SAI_NEXT_HOP_GROUP_MEMBER_ATTR_WEIGHT,
+```
+
+### Configuration and management
+This section should have sub-sections for all types of configuration and management related design. Example sub-sections for "CLI" and "Config DB" are given below. Sub-sections related to data models (YANG, REST, gNMI, etc.,) should be added as required.
+
+#### CLI Enhancements
+
+A new command is introduced to configure interfaces for MPLS.
+
+    # Enable/disable MPLS per INTERFACE.
+    config interface mpls add|remove <intf-name>
+
+#### Config DB Enhancements
+
+The existing INTERFACE table is enhanced to accept a new "mpls" enable/disable attribute.
+##### INTERFACE
+``` rfc5234
+INTERFACE|{{ifname}}
+    "mpls":{{enable|disable}} (OPTIONAL)
+
+; Defines schema for MPLS configuration attribute
+key                      = INTERFACE:ifname    ; Interface name
+; value annotations
+ifname                   = 1*64VCHAR           ; name of the Interface
+; field                  = value
+mpls                     = "enable"/"disable"  ; Enable/disable MPLS function. Default is disable
+```
+
+The existing PORTCHANNEL_INTERFACE table is enhanced to accept a new "mpls" enable/disable attribute.
+##### PORTCHANNEL_INTERFACE
+``` rfc5234
+PORTCHANNEL_INTERFACE|{{ifname}}
+    "mpls":{{enable|disable}} (OPTIONAL)
+
+; Defines schema for MPLS configuration attributes
+key                      = PORTCHANNEL_INTERFACE:ifname   ; Port Channel Interface name
+;value annotations
+ifname                   = 1*64VCHAR                      ; name of the Interface (Port Channel)
+; field                  = value
+mpls                     = "enable"/"disable"             ; Enable/disable MPLS function. Default disable
+```
+
+### Warmboot and Fastboot Design Impact
+MPLS design will not affect warmboot or fastboot design.
+
+### Restrictions/Limitations
+In this document, MPLS support is only for static lsp route support. The scope of routing stack supporting dynamic creation of MPLS tunnel is not in the design.
+
+### Testing Requirements/Design
+Using external routing controller to set up static lsp route for push/pop/swap operation on MPLS traffic and verify traffic is passing.
+
+#### Unit Test cases
+- Using Juniper cRPD set push operation lsp, observe ip traffic goes through router and egress side will have MPLS format with correct label.
+- Using Juniper cRPD set pop operation lsp, observe mpls with single label traffic goes through router and egress side will have IP packets.
+- Using Juniper cRPD set swap operation lsp, observe mpls traffic goes through router and egress side will have MPLS format with different label.
+#### System Test cases
+Not available
```

Comment:
I think we need system test cases developed, otherwise, how can the community verify the feature end-to-end?

Reply:
we are working on it.